Feature/geolocation

bodunadebiyi opened the pull request.

bodunadebiyi committed:
[feature #111961461] Create migrations
Creating a migration class to add longitude and latitude columns to the
restaurant table

bodunadebiyi committed:
[feature #111961461] Modify columns
Make the longitude and lattitude columns nullable

bodunadebiyi committed:
[feature #11196146] Modify restaurant table
write migrations to include longitude and latitude columns in the restaurant
table

bodunadebiyi committed:
[feature #111961461] Implemented first phase of geolocator
1. When a restauranteur adds a restaurant its location is converted into
longitude and latitude coordinates using the google maps api
2. The end point for fetching restaurants closeby and the distance between
these restaurants and the users present location

bodunadebiyi committed:
[feature ##111961461] second phase of geolocator
On the homepage the 'closeby feature' has been implemented. The users current
location is fetched using html5 geolocation and an ajax request is made
to the restaurant controller to fetch restaurants close by and its dynamically
loaded on the page

bodunadebiyi committed:
[feature #111961461] Geolocation tests written

bodunadebiyi committed:
[fix #111961461] Fixing circle ci tests

bodunadebiyi committed:
[fix #111961461] Fixing coveralls error

bodunadebiyi committed:
Applied fixes from StyleCI

bodunadebiyi committed:
Merge pull request #137 from andela/analysis-zEAPmm

Applied fixes from StyleCI

andela-anandaa commented:
@andela-badebiyi - fix the merge conflicts.

bodunadebiyi committed:
Merge branch 'develop' into feature/geolocation

bodunadebiyi committed:
[chore #111961461] fix merge conflict with develop branch

bodunadebiyi committed:
[feature #111961461] Updating composer.lock file

andela-anandaa commented:
@andela-badebiyi -- fix the StyleCI issues.

andela-anandaa commented on app/Http/Controllers/RestaurantController.php:
```diff
@@ -116,6 +121,117 @@ public function createAdd(Request $request)
         return redirect('restaurants');
     }
+    /**
+    public function getTest()
```
Remove this.

andela-anandaa commented on app/Http/Controllers/RestaurantController.php:
```diff
@@ -116,6 +121,117 @@ public function createAdd(Request $request)
+        // Calculate the distance in degrees using Hervasine formula
+        $degrees = rad2deg(acos((sin(deg2rad($point1_lat)) *
+            sin(deg2rad($point2_lat))) +
+        (cos(deg2rad($point1_lat)) *
```
Align this line with the rest.

andela-anandaa commented on resources/views/partials/geolocation.blade.php:
```diff
@@ -0,0 +1,78 @@
+<script>
+	$(document).ready(function(){
```
Don't mix PHP and JS. You should have a separate file JS (`.js`)

andela-anandaa commented on resources/views/partials/geolocation.blade.php:
```diff
@@ -0,0 +1,78 @@
+
+	    //store longitude and latitude coordinates in cookies
+	    function setPosition(position) 
+	    {
```
The opening bracket should be on line 11. `function nameOfFun() {`

andela-anandaa commented on resources/views/partials/geolocation.blade.php:
```diff
@@ -0,0 +1,78 @@
+	    } 
+
+	    //make an ajax request to fetch near by restaurants
+	    setTimeout(function(){
```
A space between `()` and `{`

andela-anandaa commented on resources/views/partials/geolocation.blade.php:
```diff
@@ -0,0 +1,78 @@
+
+	    //constructs the nearby restaurants elements that would be dynamically
+	    function constructHtml(input)
+	    {
```
Same to this one.

andela-anandaa commented on resources/views/partials/geolocation.blade.php:
```diff
@@ -0,0 +1,78 @@
+	});
+</script>
+<style>
+	.items{
```
Don't mix styling and PHP.

bodunadebiyi committed:
Applied fixes from StyleCI

bodunadebiyi committed:
Merge pull request #157 from andela/analysis-8jL74B

[Fix #111961461] Apply style CI fixes

bodunadebiyi committed:
[feature #111961461] resolve merge conflict with develop branch

bodunadebiyi committed:
[chore #111961461] Cleaning up code
Putting the geolocator js code in a separate js file
Removing the css from the welcome page and puttng in separate file
Removing commented out code from 'restaurant controller' file
Proper alignment of php code in 'calcDistance' method in 'restaurant' controller file
Cleaning up js code
